core/src/main/scala/org/apache/spark/util/JsonProtocol.scala

```diff
@@ -367,9 +367,15 @@ private[spark] object JsonProtocol {
         ("Job ID" -> taskCommitDenied.jobID) ~
         ("Partition ID" -> taskCommitDenied.partitionID) ~
         ("Attempt Number" -> taskCommitDenied.attemptNumber)
-      case ExecutorLostFailure(executorId, isNormalExit) =>
-        ("Executor ID" -> executorId) ~
-        ("Normal Exit" -> isNormalExit)
+      case ExecutorLostFailure(executorId, isNormalExit, reason) =>
+        if (reason.isDefined) {
+          ("Executor ID" -> executorId) ~
+          ("Normal Exit" -> isNormalExit) ~
+          ("Loss Reason" -> reason.get.toString)
```

Comment:
Nevermind, that method is for reading. (Man I really dislike this class.)

Anyway, you could probably get away with `reason.map(JString(_)).getOrElse(JNothing)` as is done in other places. Bonus brownie points for writing a helper method for that.